src/main/io/osd.c

```diff
@@ -518,17 +518,23 @@ static bool osdDrawSingleElement(uint8_t item)
         }
 
     case OSD_CRAFT_NAME:
+        // This does not strictly support iterative updating if the craft name changes at run time. But since the craft name is not supposed to be changing this should not matter, and blanking the entire length of the craft name string on update will make it impossible to configure elements to be displayed on the right hand side of the craft name.
+        //TODO: When iterative updating is implemented, change this so the craft name is only printed once whenever the OSD 'flight' screen is entered.
+
         if (strlen(pilotConfig()->name) == 0) {
             strcpy(buff, "CRAFT_NAME");
         } else {
-            for (unsigned int i = 0; i < MAX_NAME_LENGTH; i++) {
-                buff[i] = toupper((unsigned char)pilotConfig()->name[i]);
-                if (pilotConfig()->name[i] == 0) {
-                    buff[i] = SYM_BLANK;
-                }
-            }
-            buff[MAX_NAME_LENGTH] = '\0';
+            unsigned i;
+            for (i = 0; i < MAX_NAME_LENGTH; i++) {
```

Comment:
```
for (i = 0; i < MAX_NAME_LENGTH && pilotConfig()->name[i]; i++) {
  buff[i] = toupper(pilotConfig()->name[i]);
}

Reply:
This is not a good option, since it wastes screen space for the whitespace after the craft name, see #5203.

Reply:
It's equivalent, just moving the end-of-string test int `for`

Reply:
Ah yes, true, I overlooked the  `&& pilotConfig()->name[i]`. In saying that, my version makes the early exit a bit more obvious, which I prefer, and the resulting code will likely be equivalent.

Reply:
It's matter of taste ... I prefer fewer lines (easier to review / less code paths to remember) plus the fact that it's obvious that index is checked first, then NULL terminator

Reply:
Fair enough. Do you mind approving this, so it can be merged?